File: mobile/app/src/main/java/org/rotclub/area/lib/utils/BrowserUtils.java
package org.rotclub.area.lib.utils;

import static org.rotclub.area.lib.FrispyGlobalKt.BASE_URL;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.provider.Browser;

public class BrowserUtils {
    public static void openUrl(Context context, String url, String token) {
        if (url == null) {
            return;
        }
        if (!url.startsWith("http://") && !url.startsWith("https://")) {
            url = "http://" + url;
        }

        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
        Bundle bundle = new Bundle();
        bundle.putString("Authorization", "Bearer " + token);
        intent.putExtra(Browser.EXTRA_HEADERS, bundle);
        context.startActivity(intent);
    }

    public static String hrefToLink(String href) {
        if (href == null)
            return null;
        if (href.startsWith("/"))
            href = href.substring(1);
        return BASE_URL + href;
    }
}

File: mobile/app/src/main/java/org/rotclub/area/lib/utils/BrowserUtils.kt
package org.rotclub.area.lib.utils

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.provider.Browser
import org.rotclub.area.lib.baseUrl


object BrowserUtils {
    fun openUrl(context: Context, url: String, token: String) {
        var thisUrl = url
        if (!url.startsWith("http://") && !url.startsWith("https://")) {
            thisUrl = "http://$url"
        }

        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(thisUrl))
        val bundle = Bundle()
        bundle.putString("Authorization", "Bearer $token")
        intent.putExtra(Browser.EXTRA_HEADERS, bundle)
        context.startActivity(intent)
    }

    fun hrefToLink(href: String): String {
        var thisHref = href
        if (href.startsWith("/"))
            thisHref = href.substring(1)
        return baseUrl.value + thisHref
    }
}
